File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/CS/Direct2D/Paint2D/BitmapShape.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using Microsoft.WindowsAPICodePack.DirectX.Direct2D1;

namespace D2DPaint
{
    internal class BitmapShape : DrawingShape
    {
        internal RectF _rect;
        internal D2DBitmap _bitmap;
        internal float _transparency;

        internal BitmapShape(Paint2DForm parent, RectF rect, D2DBitmap bitmap, float transparency)
            : base(parent)
        {
            _rect = rect;
            _bitmap = bitmap;
            _transparency = transparency;
        }

        protected internal override void Draw(RenderTarget renderTarget)
        {
            renderTarget.DrawBitmap(_bitmap, _transparency, BitmapInterpolationMode.Linear, _rect);
        }

        protected internal override Point2F EndPoint
        {
            set
            {
                _rect.Right = Math.Max(_rect.Left + 5, value.X);
                _rect.Bottom = Math.Max(_rect.Top + 5, value.Y);
            }
        }
    }
}

File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/VB/Direct2D/Paint2D/BitmapShape.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports Microsoft.WindowsAPICodePack.DirectX.Direct2D1

Namespace D2DPaint
	Friend Class BitmapShape
		Inherits DrawingShape
		Friend _rect As RectF
		Friend _bitmap As D2DBitmap
		Friend _transparency As Single

		Friend Sub New(ByVal parent As Paint2DForm, ByVal rect As RectF, ByVal bitmap As D2DBitmap, ByVal transparency As Single)
			MyBase.New(parent)
			_rect = rect
			_bitmap = bitmap
			_transparency = transparency
		End Sub

		Protected Friend Overrides Sub Draw(ByVal renderTarget As RenderTarget)
			renderTarget.DrawBitmap(_bitmap, _transparency, BitmapInterpolationMode.Linear, _rect)
		End Sub

		Protected Friend Overrides WriteOnly Property EndPoint() As Point2F
			Set(ByVal value As Point2F)
				_rect.Right = Math.Max(_rect.Left + 5, value.X)
				_rect.Bottom = Math.Max(_rect.Top + 5, value.Y)
			End Set
		End Property
	End Class
End Namespace